src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Logging/RazorLogHubLoggerProvider.cs

```diff
@@ -1,103 +1,31 @@
-using System.Threading;
-using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Razor;
 using Microsoft.CodeAnalysis.Razor.Logging;
 using Microsoft.Extensions.Logging;
 using Microsoft.VisualStudio.Editor.Razor.Logging;
-using Microsoft.VisualStudio.Threading;
 
 namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Logging;
 
 [Shared]
 [Export(typeof(IRazorLoggerProvider))]
 internal sealed class RazorLogHubLoggerProvider : IRazorLoggerProvider
 {
-    // Unique, monotonically increasing ID to identify LogHub session to persist
-    // across server restarts.
-    private static int s_logHubSessionId;
-    private const string LogFileIdentifier = "Razor";
-
     private readonly RazorLogHubTraceProvider _traceProvider;
-    private readonly RazorLogger _razorLogger;
-    private readonly Queue<(TraceEventType Level, string Message, object[] Args)> _messageQueue;
-    private readonly CancellationTokenSource _disposalTokenSource;
-    private TraceSource? _traceSource;
 
     [ImportingConstructor]
-    public RazorLogHubLoggerProvider(RazorLogHubTraceProvider traceProvider, RazorLogger razorLogger)
+    public RazorLogHubLoggerProvider(RazorLogHubTraceProvider traceProvider)
     {
-        _traceProvider = traceProvider ?? throw new ArgumentNullException(nameof(traceProvider));
-        _razorLogger = razorLogger ?? throw new ArgumentNullException(nameof(razorLogger));
-
-        _messageQueue = new();
-        _disposalTokenSource = new();
-
-        _ = InitializeAsync(_disposalTokenSource.Token);
-    }
-
-    private async Task InitializeAsync(CancellationToken cancellationToken)
-    {
-        // Ensure that we're never on the UI thread before initialize
-        // I suspect this is overkill :D
-        await TaskScheduler.Default.SwitchTo(alwaysYield: true);
-
-        var logInstanceNumber = Interlocked.Increment(ref s_logHubSessionId);
-        var traceSource = await _traceProvider.InitializeTraceAsync(LogFileIdentifier, logInstanceNumber, cancellationToken).ConfigureAwait(false);
-        if (traceSource is null)
-        {
-            traceSource = new TraceSource("None", SourceLevels.Off) ;
-            traceSource.Listeners.Clear();
-            _razorLogger.LogError("Could not initialize trace source for Razor LogHub logger. No LogHub log will be created.");
-            return;
-        }
-
-        _traceSource = traceSource;
-
-        while (_messageQueue.Count > 0)
-        {
-            var value = _messageQueue.Dequeue();
-            if (cancellationToken.IsCancellationRequested)
-            {
-                return;
-            }
-
-            traceSource.TraceEvent(value.Level, id: 0, value.Message, value.Args);
-        }
+        _traceProvider = traceProvider;
     }
 
     public ILogger CreateLogger(string categoryName)
     {
-        return new RazorLogHubLogger(categoryName, this);
+        return new RazorLogHubLogger(categoryName, _traceProvider.GetTraceSource());
```

Comment:
So previously every time the session was reset then the trace source would get reset too, no? but now CreateLogger is passing down a TraceSource that may become invalid later. Is that right?

Reply:
MY thinking was that when a new logger is created, it always passes in the _current_ trace source. It's an attempt to set things up for potentially moving to using a new log hub file for each server instance, but that would require more changes anyway, so I don't think its a big deal.